### PR TITLE
OCPBUGS-18103: check if response is nil before using it

### PR DESCRIPTION
**Pull request by flavianmissi**

### CODE REVIEW

**Review by djslavens — APPROVED** [review]

File: pkg/storage/ibmcos/ibmcos.go [hunk 1]
```diff
@@ -804,7 +804,11 @@ func (d *driver) VolumeSecrets() (map[string]string, error) {
 		},
 	)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get resource key for service instance: %s with resp code: %d", err.Error(), resp.StatusCode)
+		respMsg := ""
+		if resp != nil {
+			respMsg = fmt.Sprintf("with resp code: %d", resp.StatusCode)
+		}
+		return nil, fmt.Errorf("unable to get resource key for service instance: %s %s", err.Error(), respMsg)
```
Comment (dmage) [inline pkg/storage/ibmcos/ibmcos.go:811]: minor - can the space be part of respMsg, i.e. `service instance: %s%s` and `respMsg = " with resp code..."?

Comment (flavianmissi) [reply pkg/storage/ibmcos/ibmcos.go]: done

**Comment by flavianmissi** [comment]
failures are unrelated
/retest

**Comment by flavianmissi** [comment]
failure seems unrelated
/retest

**Comment by flavianmissi** [comment]
failure seems unrelated
/retest

**Comment by flavianmissi** [comment]
e2e failure seems unrelated
/retest

**Pull request merged**